### DISCUSSION TITLE
我想要获取到组件的DOM实例，但是在使用ref时只能获取到内部提供的部分方法

### DISCUSSION THREAD

**Discussion by yangxiuxiu1115** [ant-design/ant-design#39762] [General]
### What problem does this feature solve?

提供组件DOM实例给使用者，部分场景下使用者可能需要获取到组件的DOM实例，例如我期望给组件添加动画，我必须完整的获取到DOM实例，但现在部分组件通过Ref只能获取到提供的方法API

### What does the proposed API look like?

可以直接在props ref中提供DOM实例

**Comment by MadCcc** [comment]
具体哪个组件需要？

**Comment by afc163** [comment]
什么组件，不建议用 ref 获取 DOM 实例，有些组件不止一个 dom 元素。

**Comment by yangxiuxiu1115** [comment]
很多组件，例如下拉菜单组件，我想要获取最外层的Dom实例，我需要给他们添加对应的动画

**Comment by yangxiuxiu1115** [comment]
现在采用的是findDomNode，但是严格模式下已经不推荐使用该API

**Comment by afc163** [comment]
建议用 id 去做绑定。

**Comment by yangxiuxiu1115** [comment]
那我应该如何将id绑定到dom实例上呢，props并不支持传入id或者classname

**Comment by afc163** [comment]
支持的吧。

**Comment by Wxh16144** [comment]
类似问题：https://github.com/ant-design/ant-design/issues/39628

**Reply by yangxiuxiu1115** [reply]
可能我期望的组件更多。

**Comment by q269384828** [comment]
外面 套一层 div 之类的也行吧, 就是可能要处理一下样式了

**Reply by yangxiuxiu1115** [reply]
我并不希望在外层嵌套div一类的元素，并且这种方式在一些布局组件 Row 和 Col 上影响是巨大的，意味着原先的布局元素可能出现问题，我可能会使用到所有的组件，我并不能确定哪些组件需要重写样式，而且这个工作对我而言不必要的
